tic-tac-toe.js

```diff
@@ -1,7 +1,132 @@
-function TicTacToe() {
+Array.prototype.allSame = function() {
+  for(var i = 1; i < this.length; i++) {
+    if (this[0] !== this[i] || this[i] === 0) {
+      return false;
+    }
+  }
+  return true;
+};
 
+function TicTacToe() {
+  this.reset();
 }
 
 TicTacToe.prototype = {
-  
-}
+
+  reset: function() {
+    $("#tic-tac-toe").empty();
+    this.makeBoard();
+    this.board = [[0, 0, 0], [0, 0, 0], [0, 0, 0]];
+    this.turn = 0;
+    this.game();
+  },
+
+  game: function() {
+    var board = this;
+    this.hoverEffect(board);
+    this.setPiece(board);
+  },
+
+  makeBoard: function() {
+    $("#tic-tac-toe").append("<table>");
+    for (var i = 0; i < 3; i++) {
+      $("table").append("<tr>");
+    }
+
+    var rows = $("table").children();
+    for (r = 0; r < 3; r++) {
+      for (c = 0; c < 3; c++) {
+        var data = $( "<td>" ).attr("id", "r" + r + "c" + c);
+        $(data).appendTo(rows[r]);
+      }
+    }
+  },
+
+  winner: function(r, c, player) {
+    var row = this.board[r];
+    var col = [this.board[0][c], this.board[1][c], this.board[2][c]];
+
+    // check row & column
+    if (row.allSame() || col.allSame()) {
+      $("#overlay").text(player + " Wins! Play again?");
+      $("#overlay").addClass("winner");
+      this.again();
+    } else if ((r + c) % 2 === 0 && this.turn !== 8) {
```

Comment:
This case is fairly opaque - I recommend commenting to clarify what's going on here.